### ISSUE TITLE
Possible issue with nonce checking

### ISSUE THREAD

**Issue by dlglin** [openwebwork/webwork2#1512]
This is based on https://webwork.maa.org/moodle/mod/forum/discuss.php?d=6367#p17322

I don't fully understand the code, but shouldn't https://github.com/openwebwork/webwork2/blob/cfb690ade3d3c55527d522ed1a8f1defd5e19f6a/lib/WeBWorK/Authen/LTIAdvanced.pm#L811 be `$db->putKey($Key);`?

Is the reason that no one else has reported this behaviour because that block of code is almost never invoked?

**Comment by taniwallach** [comment]
I agree this is a bug. There is no `put()` method to call and that is the reason for the error message:
```
Can't locate object method "put" via package "WeBWorK::DB" at /opt/webwork/webwork2/lib/WeBWorK/Authen/LTIAdvanced.pm line 811.
```

Looking at `lib/WeBWorK/DB.pm` it certainly seems that `putKey()` is what is needed here.

The code here is meant to trigger when the provided nonce shows up in the database, which should not _normally_ occur as nonces are not supposed to get reused "too soon". So yes, it should not be happening often, which is probably why it took so long for the issue to get reported.

Note: The condition under which is triggered was somewhat changed relatively recently (https://github.com/openwebwork/webwork2/commit/9966a0d78f27ac2026664b85c77af2df8eb330c0) dropping the test  `$Key->timestamp <  $self->{"timestamp"}`. I had assumed when I dropped that test that the `$self->{"timestamp"}` should essentially always be the larger value when the code got there. However, as the timestamp used is actually coming from the `$self->{oauth_timestamp}` provided by the remote LTI consumer, that is not a valid assumption.

I just made https://github.com/openwebwork/webwork2/pull/1519 to fix the bug and reinstate the conditional for when the nonce database record gets updated.
